doc/step-in-groovy.md

```diff
@@ -0,0 +1,55 @@
+# Implementing Step in Groovy
+The standard way of implementing a step is by writing Java code that extends from `Step` and `StepDescriptor`,
+but you can also define a step in Jenkins Pipeline Script.
+
+This approach is quite convenient when your step composes other
+existing steps. For example, the following `hello.groovy` defines
+a step that prints a greeting message to log:
+
+```
+String getDisplayName() {
+    return 'Greets someone'
+}
+
+void call(String person) {
+    echo "Hello ${person}";
+}
+```
+
+From Jenkins Pipeline you can use this like you'd use any other steps:
+
+```
+hello 'Duke'    // this will print 'Hello Duke'
+```
+
+`src/main/webapp/WEB-INF/steps` is the content root for Groovy files
+to be run by the CPS interpreter (aka Pipeline classpath). You can create regular package
+structures in this content root directory. For example, the above
+helloWorld.groovy could be in `src/main/webapp/WEB-INF/steps/org/kohsuke/jenkins/helloWorld.groovy`
+
+It is also possible to put other supporting classes in this content root directory.
```

Comment:
But not resources, I suppose. At worst these could always be loaded from the plugin’s regular `src/main/resources/` by

```java
Jenkins.instance.pluginManager.uberClassLoader.getResource('org/jenkins-ci/plugins/whatever/something.xml')
```

or similar.